client/swift.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2025 SAP SE
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"math"
+	"net/http"
+	"time"
+
+	"github.com/gophercloud/gophercloud/v2"
+	"github.com/gophercloud/gophercloud/v2/openstack"
+	"github.com/gophercloud/utils/v2/env"
+	"github.com/majewsky/schwift/v2"
+	"github.com/majewsky/schwift/v2/gopherschwift"
+	"github.com/sapcc/go-bits/logg"
+	"github.com/sapcc/go-bits/osext"
+)
+
+// SwiftExporter handles exporting data to Swift
+type SwiftExporter struct {
+	container *schwift.Container
+	format    string
+	filename  string
+	segSize   uint64
+}
+
+// initializeSwiftContainer creates and initializes a Swift container
+func initializeSwiftContainer(ctx context.Context, provider *gophercloud.ProviderClient, containerName string) (*schwift.Container, error) {
+	client, err := openstack.NewObjectStorageV1(provider, gophercloud.EndpointOpts{
+		Region: osext.GetenvOrDefault("OS_REGION_NAME", env.Getenv("OS_REGION_NAME")),
```

Comment:
I didn't check if you are using `NewProviderClient` but that already has OS_REGION_NAME here https://github.com/sapcc/go-bits/blob/f83e86ae829e71a18dd287144795818e53b7a247/gophercloudext/auth.go#L157